From e771f7bb5512b959e988dbeee919d7f4403fcdb9 Mon Sep 17 00:00:00 2001
From: sjegadeesan <sjegadeesan@maxlinear.com>
Date: Fri, 13 Dec 2024 11:47:42 +0530
Subject: [PATCH] WLANRTSYS-86804 6G ssid change is not reflected in OSS build

Problem is that MBSS beacon update has been handled only for
proprietary build.

This has been fixed by extending the fix to open source build
as well.
---
 hostapd/mxl_ctrl_iface.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/hostapd/mxl_ctrl_iface.c b/hostapd/mxl_ctrl_iface.c
index 05d1ff6d2..d0d22b169 100644
--- a/hostapd/mxl_ctrl_iface.c
+++ b/hostapd/mxl_ctrl_iface.c
@@ -3783,18 +3783,17 @@ int mxl_hostapd_ctrl_iface_set_beacon(struct hostapd_data *hapd)
 {
 	enum hostapd_iface_state istate = hapd->iface->state;
 
+	if ((istate == HAPD_IFACE_ENABLED)
 #ifdef CONFIG_VENDOR_MXL_PROPRIETARY
-	if ((istate == HAPD_IFACE_ENABLED) || hapd->iface->mxl_iface.acs_bgscan) {
+	    || (hapd->iface->mxl_iface.acs_bgscan)
+#endif
+	) {
 		if (hapd->iconf->multibss_enable)
 			return mxl_hostapd_ctrl_iface_update_mbss_beacon(hapd);
 		else
 			return ieee802_11_set_beacon(hapd);
 	}
 
-#else
-	return ieee802_11_set_beacon(hapd);
-#endif
-
 	wpa_printf(MSG_WARNING, "%s: set beacon in interface state %s not allowed",
 		hapd->conf->iface, hostapd_state_text(istate));
 	return -1;
-- 
2.43.0

